codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/auth/factories/HttpBasicAuthSchemeFactory.java

```diff
@@ -37,13 +38,14 @@ public static AuthScheme build() {
 
         ClientConfig identityResolverConfig = ClientConfig.builder()
                 .name("http_login_identity_resolver")
-                .type(Hearth.IDENTITY_RESOLVER.toString())
                 .documentation(
                         identityResolverDocumentation.formatted(
                                 Hearth.IDENTITY_RESOLVER,
                                 identityType,
                                 HttpBasicAuthTrait.ID))
+                .documentationType(Hearth.IDENTITY_RESOLVER.toString())
                 .defaultDynamicValue(defaultConfigValue)
+                .constraint(new TypeConstraint(Hearth.IDENTITY_RESOLVER.toString()))
```

Comment:
I'd still lean towards just adding a convince builder method for `validateType`. 

Another convenience builder method we could add: `validationDocumentationAndRbsType(oneTypeForAll)`

Most config we have follows this - and its better in those cases to ensure that they all match and are correct.